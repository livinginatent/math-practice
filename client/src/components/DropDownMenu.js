import * as React from "react";
import Button from "@mui/material/Button";
import Menu from "@mui/material/Menu";
import MenuItem from "@mui/material/MenuItem";
import MenuIcon from "@mui/icons-material/Menu";
import { IconButton } from "@mui/material";
import UserDashboard from "../Pages/UserDashboard";
import { useNavigate } from "react-router";
import { Link } from "react-router-dom";

export default function DropDownMenu(props) {
  const [anchorEl, setAnchorEl] = React.useState(null);
  const open = Boolean(anchorEl);
  const handleClick = (event) => {
    setAnchorEl(event.currentTarget);
  };
  const handleClose = () => {
    setAnchorEl(null);
  };
  if (props.navLocation === "dashboard") {
    return (
      <div>
        <IconButton
          startIcon={<MenuIcon />}
          id="basic-button"
          aria-controls={open ? "basic-menu" : undefined}
          aria-haspopup="true"
          aria-expanded={open ? "true" : undefined}
          onClick={handleClick}
        >
          <MenuIcon />
        </IconButton>
        <Menu
          id="basic-menu"
          anchorEl={anchorEl}
          open={open}
          onClose={handleClose}
          MenuListProps={{
            "aria-labelledby": "basic-button",
          }}
        >
          <Link style={{ textDecoration: "none", color: "inherit" }} to="/profile">
            <MenuItem onClick={handleClose}>Profile</MenuItem>
          </Link>
          <Link style={{ textDecoration: "none", color: "inherit" }} to="/">
            <MenuItem onClick={handleClose}>Home</MenuItem>
          </Link>
        </Menu>
      </div>
    );
  } else if(props.props.navLocation ==='profile'){
    return (
      <div>
        <IconButton
          startIcon={<MenuIcon />}
          id="basic-button"
          aria-controls={open ? "basic-menu" : undefined}
          aria-haspopup="true"
          aria-expanded={open ? "true" : undefined}
          onClick={handleClick}
        >
          <MenuIcon />
        </IconButton>
        <Menu
          id="basic-menu"
          anchorEl={anchorEl}
          open={open}
          onClose={handleClose}
          MenuListProps={{
            "aria-labelledby": "basic-button",
          }}
        >
          <Link
            style={{ textDecoration: "none", color: "inherit" }}
            to="/"
          >
            <MenuItem onClick={handleClose}>Home</MenuItem>
          </Link>
          <Link style={{ textDecoration: "none", color: "inherit" }} to="/dashboard">
            <MenuItem onClick={handleClose}>Dashboard</MenuItem>
          </Link>
        </Menu>
      </div>
    );
  }
   else{

    return (<div>
      <IconButton
        startIcon={<MenuIcon />}
        id="basic-button"
        aria-controls={open ? "basic-menu" : undefined}
        aria-haspopup="true"
        aria-expanded={open ? "true" : undefined}
        onClick={handleClick}
      >
        <MenuIcon />
      </IconButton>
      <Menu
        id="basic-menu"
        anchorEl={anchorEl}
        open={open}
        onClose={handleClose}
        MenuListProps={{
          "aria-labelledby": "basic-button",
        }}
      >
        <Link
          style={{ textDecoration: "none", color: "inherit" }}
          to="/profile"
        >
        <MenuItem onClick={handleClose}>Profile</MenuItem>
          
        </Link>
        <Link
          style={{ textDecoration: "none", color: "inherit" }}
          to="/dashboard"
        >
        <MenuItem onClick={handleClose}>Dashboard</MenuItem>
          
        </Link>
      </Menu>
    </div>
)  }
}
