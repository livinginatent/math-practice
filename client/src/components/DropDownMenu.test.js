import * as React from "react";
import renderer from "react-test-renderer";
import DropDownMenu from "./DropDownMenu";

describe("DropDownMenu component", () => {
  it("renders correctly for the dashboard nav location", () => {
    const tree = renderer
      .create(<DropDownMenu  dashLocation="dashboard" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly for the profile nav location", () => {
    const tree = renderer
      .create(<DropDownMenu profileLocation="profile" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly for the default nav location", () => {
    const tree = renderer.create(<DropDownMenu />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
