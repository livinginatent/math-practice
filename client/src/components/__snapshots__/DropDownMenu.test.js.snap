// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDownMenu component renders correctly for the dashboard nav location 1`] = `
<div>
  <button
    aria-haspopup="true"
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    disabled={false}
    id="basic-button"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    startIcon={<Memo />}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="MenuIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
</div>
`;

exports[`DropDownMenu component renders correctly for the default nav location 1`] = `
<div>
  <button
    aria-haspopup="true"
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    disabled={false}
    id="basic-button"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    startIcon={<Memo />}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="MenuIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
</div>
`;

exports[`DropDownMenu component renders correctly for the profile nav location 1`] = `
<div>
  <button
    aria-haspopup="true"
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    disabled={false}
    id="basic-button"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    startIcon={<Memo />}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="MenuIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
